Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -482,6 +486,20 @@ static HRESULT _createPathReadyForFigure(CGPathRef previousPath,
     return S_OK;
 }
 
+static const CGFloat sc_zeroAngleThreshold = .00001;
+// This function will return a normalized angle in radians between 0 and 2pi. This is to standardize
+// the calculations for arcs since 0, 2pi, 4pi, etc... are all visually the same angle.
+static CGFloat __normalizeAngle(CGFloat originalAngle) {
+    CGFloat returnAngle = fmod(originalAngle, 2 * M_PI);
```

Comment:
Not quite, since that would ignore any angle like 7pi/2 and instead return 2 *M_PI. We only want to return an angle of 2pi if the angle is a multiple of 2pi and not 0. That's what the extra checks are for.